sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/FeatureFlagConfigurationSetting.cs

```diff
@@ -58,14 +63,28 @@ internal FeatureFlagConfigurationSetting()
         /// <param name="featureId">The identified of the feature flag.</param>
         /// <param name="isEnabled">The value indicating whether the feature flag is enabled.</param>
         /// <param name="label">A label used to group this configuration setting with others.</param>
-        public FeatureFlagConfigurationSetting(string featureId, bool isEnabled, string label = null): this()
+        public FeatureFlagConfigurationSetting(string featureId, bool isEnabled, string label = null): this(featureId, isEnabled, label, default)
+        {
+        }
+
+        /// <summary>
+        /// Initializes an instance of the <see cref="FeatureFlagConfigurationSetting"/> using a provided feature id and
+        /// the enabled value.
+        /// </summary>
+        /// <param name="featureId">The identified of the feature flag.</param>
+        /// <param name="isEnabled">The value indicating whether the feature flag is enabled.</param>
+        /// <param name="label">A label used to group this configuration setting with others.</param>
+        /// <param name="etag">The ETag value for the configuration setting.</param>
+        public FeatureFlagConfigurationSetting(string featureId, bool isEnabled, string label, ETag etag) : this()
```

Comment:
Curious: why do callers need to be able to create a new instance with an ETag?  Shouldn't ETag be populated by retrieving it from the service and therefore only needed by internal serialization constructors?

Reply:
In a nutshell, it is needed for a "GetSetting" overload that follows an odd pattern.  Please see the referenced bug for context.